src/main/java/utility/StringDelimiters.java

```diff
@@ -0,0 +1,46 @@
+package utility;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+
+final class StringDelimiters {
+
+	private static final String REGEX_DELIMITER = "|";
+
+	private final Collection<String> delimiters;
+
+	StringDelimiters(Collection<String> delimiters) {
+		validate(delimiters);
+		this.delimiters = delimiters;
+	}
+
+	public static StringDelimiters of(String delimiter, String... delimiters) {
+		HashSet<String> delimiterSet = new HashSet<>(Arrays.asList(delimiters));
+		delimiterSet.add(delimiter);
+		return new StringDelimiters(delimiterSet);
+	}
```

Comment:
아래와 같이 파라미터를 하나로 합쳐볼 수 있을 것 같아요!

```suggestion
	public static StringDelimiters of(String... delimiters) {
		return new StringDelimiters(Arrays.asList(delimiters));
	}
```

테스트를 단단히 만들어주셔서,
이렇게 리펙토링을 두려움없이 할 수 있게 짜신 것 정말 👍입니다

Reply:
코멘트 감사합니다!
사실 말씀해주셨던 것 처럼 처음에는 붙어있었습니다만
컴파일 레벨에서 한개 이상의 인자를 강제하고 싶어서 이렇게 변형하게 되었습니다.
이런 경우 하나의 인자로 받고 내부에서 검증로직을 추가하는편이 더 깔끔해질까요?

Reply:
아하, 그런 의도가 있었군요!

넵 저는 그 편이 더 깔끔하고, 개발자 간 의사소통 시에도 오해의 여지가 없을 것 같아요!

```
	private void validate(Collection<String> delimiters) {
		for (String delimiter : delimiters) {
			checkNull(delimiter);
		}
	}
```
동일 클래스 내 위 부분에 `delimiters`의 empty여부를 체크해볼 수도 있겠네요!

Reply:
넵 의견 감사합니다!!